neon_mq_connector/connector.py

```diff
@@ -393,35 +424,37 @@ def run_consumers(self, names: tuple = (), daemon=True):
         if not names or len(names) == 0:
             names = list(self.consumers)
         for name in names:
-            if name in list(self.consumers) and not self.consumers[name].is_alive():
+            if isinstance(self.consumers.get(name), ConsumerThread) and not self.consumers[name].is_alive():
                 self.consumers[name].daemon = daemon
                 self.consumers[name].start()
+                self.consumer_properties[name]['started'] = True
 
     def stop_consumers(self, names: tuple = ()):
         """
-        Stops consumer threads based on the name if present
-        (stops all of the declared consumers by default)
+            Stops consumer threads based on the name if present
+            (stops all of the declared consumers by default)
         """
         if not names or len(names) == 0:
             names = list(self.consumers)
         for name in names:
             try:
                 if name in list(self.consumers):
-                    self.consumers[name].join()
-                    self.consumers.pop(name)
+                    self.consumers[name].join(timeout=self.__consumer_join_timeout__)
```

Comment:
Consider logging or raising if the thread is left running

Reply:
> Consider logging or raising if the thread is left running

Fixed that